src/codeu/chat/client/simplegui/NewAccount.java

```diff
@@ -0,0 +1,98 @@
+package codeu.chat.client.simplegui;
+
+import java.awt.Button;
+import java.awt.GridBagConstraints;
+import java.awt.GridBagLayout;
+import java.awt.event.ActionEvent;
+import java.awt.event.ActionListener;
+import java.util.HashMap;
+
+import javax.swing.BoxLayout;
+import javax.swing.JButton;
+import javax.swing.JFrame;
+import javax.swing.JLabel;
+import javax.swing.JPanel;
+import javax.swing.JPasswordField;
+import javax.swing.JTextField;
+
+import codeu.chat.UserMap;
+
+@SuppressWarnings("serial")
+/**
+ * @author  Troy Jones
+ * @date    5/21/17
+ * @brief   Contains all GUI necessary to execute the frame to 
+ *          create a new account.
+ */
+public class NewAccount extends JFrame {
+
+    UserMap userMap;
+    HashMap<String, String> usernameAndPassword;
+
+    public NewAccount (JFrame mainFrame) {
+        JFrame newAccountFrame = new JFrame();
+        userMap = new UserMap();
+        // create mainPanel
+        JPanel mainPanel = new JPanel();
+        BoxLayout mainPanelLayout = 
+                new BoxLayout(mainPanel, BoxLayout.PAGE_AXIS);
+        mainPanel.setLayout(mainPanelLayout);
+
+        // fill welcomeTextPanel
+        JPanel welcomeTextPanel = new JPanel();
+        JLabel welcomeLabel = new JLabel("Welcome to ChatU!");
+        welcomeTextPanel.add(welcomeLabel);
+        mainPanel.add(welcomeTextPanel);
+
+        // username label and textfield
+        JPanel usernamePanel = new JPanel();
+        JLabel usernameLabel = new JLabel("Choose a user name");
+        usernamePanel.add(usernameLabel);
+        JTextField userNameTextField = new JTextField(10);
+        usernamePanel.add(userNameTextField);
+        mainPanel.add(usernamePanel);
+
+        // password label and textfield
+        JPanel userPasswordPanel = new JPanel();
+        JLabel userPasswordLabel = new JLabel("Choose a password");
+        userPasswordPanel.add(userPasswordLabel);
+        JPasswordField userpasswordTextField = new JPasswordField(10);
+        userPasswordPanel.add(userpasswordTextField);
+        mainPanel.add(userPasswordPanel);
+
+        // fill buttonsPanel
+        JPanel buttonsPanel = new JPanel();
+        JButton createAccountButton = new JButton("Create");
+        createAccountButton.addActionListener(new ActionListener() {
+            public void actionPerformed(ActionEvent ae) {
+
+                String userName = new String(userNameTextField.getText());
+                usernameAndPassword = userMap.getMap();
```

Comment:
This should be a method inside UserMap

i.e. a method called doesUserExist(String username) that returns true/false
and a method called addUser(String username, String password) that allows you to add the user to the map